cogs/highlight.py

```diff
@@ -198,8 +209,18 @@ def _is_registered(self, guild_idx, guild_id, user_id):
             notify_msg += "[{0}] {1.author.name}#{1.author.discriminator}: {1.content}\n".format(time,msg)
         await self.bot.send_message(user,notify_msg)
         
-    async def _last_spoken(self):
-        pass # TODO: check if user sent message in last 10 seconds or so, if so dont send a PM
+    async def _is_active(self, user_id, channel, message):
+        # NOTE: this is a naive approach to checking activity, for now to keep simple, just see if user
+        # created a message in the last 50 messages. and if they did, was in less than 20 seconds since 
+        # the message we are currently checking for highlight words

```

Comment:
@Injabie3 yeah i was initially going to do that, but the issue with keeping a dict is that this active check is per CHANNEL, so that means i would need to store their last post per every channel in every server. that would just be a pain and use a lot of memory. this method is called everytime in `check_highlights` so it is being activated every `on_message`. Main reason why I want to keep it in some form close to this implementation is cuz it's light on memory. still not 100% sure about just getting last 50 logs and how stable that is, will need to think about that a bit more to see if there is a more reliable option